### ISSUE TITLE
Classifier for predicting when a math function will error

### ISSUE THREAD

**Issue by jpfairbanks** [jpfairbanks/SemanticModels.jl#81]
For the validation functionality, we want to show that we can learn to predict when a model will fail based on traces of its execution.

Given a function `f`
```julia
function f(x)
    a = sum(x.^2 .- x)
    s = sqrt(a)
     return s
end
```
Can we build a model that can determine if `f(x)` will return successfully or fail?
This seems like an oversimplified problem (why don't you just run it) but we want to generalize this to cases where the definition is more subtle, and perhaps only definable by a human expert looking at plots.

_Steps_:
1. Generate input and output pairs of `x`, and `f(x)`.
1. Use the code in `test/transform/varextract.jl` to generate traces of `f`
1. Build a model on input output pairs to see how many data points you need to reliably predict failure
1. Build a model on traces to predict failure
1. Compare the number of samples.

The model that uses traces should need fewer examples, because there is information in the trace that is not present in just the input and output pair. A trace is a very high dimensional feature extractor that is based on the implementation details of the functions that evaluate the math. 

I'd like to see a notebook that did everything in the _Steps_ as a pilot study we can add to the docs for the next release.

**Comment by infvie** [comment]
I think that another consideration or assumption that hasn't been made clear to me is if we expect all the scripts we are going to be digesting are relatively inexpensive to actually run? If a scientist wants to work on improving a script that might take a week to run the novelty of having a kg of the program which would take a week to build may not be worth it. How are we handling this? Are we expecting small fast running test cases? @jpfairbanks

**Comment by jpfairbanks** [comment]
Yeah, we expect to get the knowledge graph from running on small data sets which will be fast. The structure of the program shouldn't change when you scale up the input size.

**Comment by jpfairbanks** [comment]
So our first cut at this failed, most likely because the straight up trace of the program when linearized requires ultra long range phenomena to make sense. We are trying a new approach based on recursive neural nets.
